src/main/java/org/testng/internal/PackageUtils.java

```diff
@@ -83,68 +84,92 @@ public static void addClassLoader(final ClassLoader loader) {
       }
     }
 
+    ParamContainer container = new ParamContainer(included,excluded,vResult);
```

Comment:
@juherr Yes I understand that. I was thinking of doing it in a phased manner. So I didn't want to change things on the fly, but still try to keep it controlled. the current code was basically having a lot of nesting. My intent was to break the nesting alone. I wasn't looking at refactoring it internally (atleast not with this Pull request)